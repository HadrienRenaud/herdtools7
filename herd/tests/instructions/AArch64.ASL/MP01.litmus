AArch64 MP01
{
0:X7=x; 0:X8=y;
1:X7=x; 1:X8=y;
}
  P0              |  P1          ;
 LDR W1,[X7]      | MOV W0,#1    ;
 CMP W1,#0        | STR W0,[X8]  ;
 LDR W3,[X8]      | STLR W0,[X7] ;

exists (0:X1=1 /\ 0:X3=0)
