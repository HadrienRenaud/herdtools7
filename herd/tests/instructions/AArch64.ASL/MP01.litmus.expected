Test MP01 Allowed
States 4
0:X1=0; 0:X3=0;
0:X1=0; 0:X3=1;
0:X1=1; 0:X3=0;
0:X1=1; 0:X3=1;
Ok
Witnesses
Positive: 1 Negative: 3
Condition exists (0:X1=1 /\ 0:X3=0)
Observation MP01 Sometimes 1 3
Hash=f6e8d163288dce9e9b25d40d69f01a3c

